File: examples/nested-config/src/main/java/com/riversoforion/lena/example/NetworkConfig.java
/*
 * Copyright (c) 2024-2025. Eric McIntyre / Rivers of Orion
 */
package com.riversoforion.lena.example;

import org.riversoforion.lena.config.ConfigurationProperties;
import org.riversoforion.lena.config.Name;
import org.riversoforion.lena.config.Namespace;

import static org.riversoforion.lena.config.ConfigurationSources.*;

public class NetworkConfig extends ConfigurationProperties {

    public NetworkConfig(Namespace ns) {

        super(prioritized(forEnvironment(), forSystemProperties()), ns);
        returnNullForMissing();
    }

    public long readTimeout() {

        return longVal(Name.of("read/timeout"));
    }

    public long connectTimeout() {

        return longVal(Name.of("connect/timeout"));
    }
}

File: examples/nested-config/src/main/kotlin/com/riversoforion/lena/example/NetworkConfig.kt
/*
 * Copyright (c) 2024-2026. Eric McIntyre / Rivers of Orion
 */
package com.riversoforion.lena.example

import org.riversoforion.lena.config.ConfigurationProperties
import org.riversoforion.lena.config.ConfigurationSource
import org.riversoforion.lena.config.Namespace

class NetworkConfig(source: ConfigurationSource, ns: Namespace) : ConfigurationProperties(source, ns) {

    val readTimeout: Long by long("read", "timeout", default = 5_000L)
    val connectionTimeout: Long by long("connection", "timeout", default = 5_000L)
}
